mapie/metrics.py

```diff
@@ -307,3 +311,441 @@ def top_label_ece(
+    >>> y_true = np.array([5, 7.5, 9.5])
+    >>> y_intervals = np.array([
+    ... [[4, 4], [6, 7.5]],
+    ... [[6.0, 8], [9.0, 10]],
+    ... [[9, 9], [10.0, 10.0]]
+    ... ])
+    >>> print(regression_ssc_score(y_true, y_intervals, num_bins=2))
+    [1.  0.5]
+    """
+    return np.min(regression_ssc(y_true, y_intervals, num_bins), axis=1)
+
+
+def classification_ssc(
+    y_true: NDArray,
+    y_pred_set: NDArray,
+    num_bins: Union[int, None] = None
+) -> NDArray:
+    """
+    Compute Size-Stratified Coverage metrics proposed in [3] that is
+    the conditional coverage conditioned by the size of the predictions sets.
+    The sets are ranked by their size (ascending) and then divided into
+    num_bins groups: one value of coverage by groups is computed.
+
+    [3] Angelopoulos, A. N., & Bates, S. (2021).
+    A gentle introduction to conformal prediction and
+    distribution-free uncertainty quantification.
+    arXiv preprint arXiv:2107.07511.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_pred_set: NDArray of shape (n_samples, n_class, n_alpha)
+    or (n_samples, n_class)
+        Prediction sets given by booleans of labels.
+    num_bins: int or None
+        Number of groups. If None, one value of coverage by possible
+        size of sets (n_classes +1) is computed.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha, num_bins)
+
+    Raises
+    ------
+    ValueError
+        If num_bins is greater than n_samples.
+
+    Examples
+    --------
+    >>> from mapie.metrics import classification_ssc
+    >>> import numpy as np
+    >>> y_true = y_true_class = np.array([3, 3, 1,])
+    >>> y_pred_set = np.array([
+    ... [False, False, True, True],
+    ... [False, True, False, True],
+    ... [False, True, True, False]
+    ... ])
+    >>> print(classification_ssc(y_true,y_pred_set,num_bins=2))
+    [[1. 1.]]
+    """
+    y_true = cast(NDArray, column_or_1d(y_true))
+    y_pred_set = check_array_shape_classification(y_true, y_pred_set)
+
+    sizes = np.sum(y_pred_set, axis=1)
+    if num_bins is not None:
+        check_number_bins(num_bins)
+        if num_bins > y_pred_set.shape[0]:
+            raise ValueError(
+                "Please provide a bin number lower than the number of samples"
+            )
+
+        indexes_sorted = np.argsort(sizes, axis=0)
+        indexes_bybins = np.array_split(indexes_sorted, num_bins, axis=0)
+        coverages = np.zeros((y_pred_set.shape[2], num_bins))
+        for i, indexes in enumerate(indexes_bybins):
+            coverages[:, i] = classification_coverage_score_v2(
+                y_true[indexes],
+                np.take_along_axis(
+                    y_pred_set, np.expand_dims(indexes, axis=1), axis=0
+                )
+            )
+    else:
+        n_classes = y_pred_set.shape[1]
+        coverages = np.zeros((y_pred_set.shape[2], n_classes + 1))
+        for alpha in range(y_pred_set.shape[2]):
```

Comment:
why do you need to loop over the alphas here and not in your first "if" condition ?

Reply:
Because in the if condition the samples are splits with np.array_split() that returns a fixed number of sample per bin regardless of alpha. In the case num_bins=None, we create a split for each possible size of set : it is possible to have for alpha1 x sets of size 1 and y sets of size 1 for alpha2 (x!=y). In this case, the bins have not the same size for each alpha so we need a for loop.

Reply:
so it means that when we give a number of bins equal to 10 for instance, two observations with the same number of labels car belong to two different bins ?

Reply:
I have deleted the if case, the two cases (num_bins=None and num_bins=int) are computed the same way. (It also solves the problem where two sets of the same size could have been in differents bins)